### ISSUE TITLE
IPv6 neighbour proxy support

### ISSUE THREAD

**Issue by daaku** [docker/docker#10919]
Hey, I was working towards neighbour proxy support for IPv6 environments (https://tools.ietf.org/html/rfc4389). Specifically I need to add the global address allocated to the container as a neighbour proxy on the host. In theory I could build this using events, though there may be a race if the container wants external network right away. The better solution would be to add this to docker itself so when an IP is allocated we (optionally) add a neighbour entry, and similarly remove it when the IP is released.

Thoughts if this is something docker itself should have support for?

**Comment by MalteJ** [comment]
@daaku do you know http://docs.master.dockerproject.com/articles/networking/#using-ndp-proxying ?

**Comment by daaku** [comment]
Yep, I was wondering if implementing the logic to perform the equivalent of `ip -6 neigh add proxy 2001:db8::c009 dev eth0` (and the corresponding removal) within docker itself be welcome. Since docker makes up the IPv6 address, if it also adds the proxy entry before the container is started, it would make networking work as soon as the container starts up.

**Comment by jansauer** [comment]
:+1:

**Comment by thaJeztah** [comment]
ping @mrjana @mavenugo not sure what's being asked here, perhaps you can have a look if this needs changes in libnetwork

**Comment by CtrlZvi** [comment]
I recently started looking into this. It looks like it would need changes to the bridge driver in libnetwork. I'm not sure if neighbor proxy makes sense for any of the other drivers.

**Comment by LK4D4** [comment]
I think @CtrlZvi lost interest. Leaving it open though.

**Comment by mostolog** [comment]
I think it's awaiting for review: https://github.com/docker/libnetwork/pull/1714
